tripal/tests/src/Kernel/Entity/TripalEntityFieldTest.php

```diff
@@ -0,0 +1,180 @@
+   *   The label of the scenario in the YAML.
+   */
+  public function testTripalEntitySaveTitleUrlBasic(int $current_scenario_key, string $current_scenario_label) {
+    $current_scenario = $this->scenarios[$current_scenario_key];
+    $this->assertEquals($current_scenario_label, $current_scenario['label'], "We may not have retrieved the expected scenario as the labels did not match.");
+
+    // 1. Create the entity with that value set.
+    $submitted_title = $this->randomString();
+    $entity = TripalEntity::create([
+      'title' => $submitted_title,
+      'type' => $this->bundle_name,
+    ] + $current_scenario['create']['user_input']);
+    $this->assertInstanceOf(TripalEntity::class, $entity, "We were not able to create a piece of tripal content to test our " . $current_scenario['label'] . " scenario.");
+    $status = $entity->save();
+    $this->assertEquals(SAVED_NEW, $status, "We expected to have saved a new entity for our " . $current_scenario['label'] . " scenario.");
+
+    // 2. Load the entity we just created so we can check the values.
+    $created_entity = TripalEntity::load($entity->id());
+    $this->assertFieldValuesMatch($current_scenario['create']['expected'], $created_entity, '"' . $current_scenario['label'] . '" being created. ');
+    // -- Title.
+    $this->assertNotEquals($submitted_title, $created_entity->getTitle(), "The submitted title should never be used but it was when CREATING the entity for the '" . $current_scenario['label'] . "' scenario.");
+    $this->assertEquals($current_scenario['create']['title'], $created_entity->getTitle(), "We did not get the title we expected when CREATING the entity for the '" . $current_scenario['label'] . "' scenario.");
+    // -- URL.
+    $retrieved_alias = $created_entity->getAlias();
+    $this->assertIsArray($retrieved_alias, "The retrieved path should be an array when CREATING the entity for the '" . $current_scenario['label'] . "' scenario.");
+    $this->assertArrayHasKey('alias', $retrieved_alias, "The retrieved path should have an alias property when CREATING the entity for the '" . $current_scenario['label'] . "' scenario.");
+    $this->assertEquals($current_scenario['create']['url'], $retrieved_alias['alias'], "We did not get the url alias we expected when CREATING the entity for the '" . $current_scenario['label'] . "' scenario.");
+
+    // 3. Make changes and then save again.
+    foreach ($current_scenario['edit']['user_input'] as $field_name => $new_values) {
+      $created_entity->set($field_name, $new_values);
+    }
+    // @debug print_r($created_entity->toArray());
+    $status = $created_entity->save();
+    $this->assertEquals(SAVED_UPDATED, $status, "We expected to have updated the existing entity for our " . $current_scenario['label'] . " scenario.");
+
+    // 4. Load the entity we just updated so we can check the values.
+    $updated_entity = TripalEntity::load($created_entity->id());
+    // @debug print_r($updated_entity->toArray());
```

Comment:
What is the preference for leaving commented out debugging code like this in for the final PR? 
```suggestion
    $status = $created_entity->save();
    $this->assertEquals(SAVED_UPDATED, $status, "We expected to have updated the existing entity for our " . $current_scenario['label'] . " scenario.");

    // 4. Load the entity we just updated so we can check the values.
    $updated_entity = TripalEntity::load($created_entity->id());
```

Reply:
My preference is to leave it in with the `@debug` tag in automated tests --sparingly but to remove it entirely from the main codebase (i.e. not test-related files). I'm most likely to leave it in if it's debugging for an object where it might not be obvious how to print since you cant print_r() it directly due to size.

Reply:
Okay sounds good